### PR TITLE
cosmetic(pie-docs): sticky nav and scroll-lock body on mobile nav open

### PR DESCRIPTION
**Pull request by jamieomaguire**
## Describe your changes (can list changeset entries if preferable)
[Changed] - Made the navigation sticky on desktop and scroll-locked on mobile

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [x] If it is a `PIE Docs` change, I have reviewed the PR preview
- [x] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [x] If it is a `PIE Docs` change, I have reviewed the PR preview
- [x] If there are visual test updates, I have reviewed them

### CODE REVIEW

**Review by FayeCarter — APPROVED** [review]

File: apps/pie-docs/src/assets/styles/components/_navToggle.scss [hunk 1]
```diff
@@ -29,6 +29,15 @@
     }
 }
 
+html:has(.c-navToggle-input:checked) body {
+    /* when modal active */
```
Comment (ashleynolan) [inline apps/pie-docs/src/assets/styles/components/_navToggle.scss:33]:
```suggestion
    /* when navigation overlay is active */
```

File: apps/pie-docs/src/assets/styles/components/_navToggle.scss [hunk 2]
```diff
@@ -29,6 +29,15 @@
...
+    touch-action: none;
+    -webkit-overflow-scrolling: none;
+    overflow: hidden;
+    /* Other browsers */
```
Comment (ashleynolan) [inline apps/pie-docs/src/assets/styles/components/_navToggle.scss:37]: What does `other browsers` refer to here? I'd guess the -webkit property above is chrome and the property below is firefox/safari?

If that's the case, I'd prob mark this as:

```suggestion
    /* non webkit browsers */
    overscroll-behavior: none;
```

**Review by ashleynolan — COMMENTED** [review]
Couple of super small comments, otherwise LGTM

**Comment by ashleynolan** [comment]
Reviewing the Percy checks, you'll likely need to rebase against master (it looks like it's flagging some changes to the engineering content which I merged earlier)

File: apps/pie-docs/src/assets/styles/components/_nav.scss [hunk 3]
```diff
@@ -94,3 +94,16 @@ $nav-item-left-margin: calc($nav-category-gap + $nav-category-icon-width);
     border-radius: f.$radius-rounded-b;
   }
 }
+
+.c-nav-inner {
+  position: sticky;
+  top: 0;
+  bottom: 0;
+  // This is to prevent the ios safari address bar from covering part of the nav in mobile
+  max-height: calc(100vh - 170px);
```
[the remark this answers is not in the document]
Comment (jamieomaguire) [reply apps/pie-docs/src/assets/styles/components/_nav.scss:103]: yeah happy to, honestly this was just trial and error. Unsure if there's an established height I could use instead

**Review by FayeCarter — DISMISSED** [review]

**Review by FayeCarter — APPROVED** [review]

File: apps/pie-docs/src/assets/styles/components/_nav.scss [hunk 4]
```diff
@@ -94,3 +94,16 @@ $nav-item-left-margin: calc($nav-category-gap + $nav-category-icon-width);
     border-radius: f.$radius-rounded-b;
   }
 }
+
+.c-nav-inner {
+  position: sticky;
+  top: 0;
+  bottom: 0;
+  // This is to prevent the ios safari address bar from covering part of the nav in mobile
+  max-height: 75vh;
```
Comment (ashleynolan) [inline apps/pie-docs/src/assets/styles/components/_nav.scss:103]: Think this is absolutely fine to get in for launch - would be good to make a ticket to clean up though, as it does look a bit strange when scrolling without the address bar covering it (so would be good to fix it down the line).

Comment (jamieomaguire) [reply apps/pie-docs/src/assets/styles/components/_nav.scss:103]: cheers - fully agree! I've created a ticket :)

**Review by raoufswe — APPROVED** [review]

**Pull request merged**